### ISSUE TITLE
Realm localization: Inconsistent message resolving regarding language fallbacks for different themes

### ISSUE THREAD

**Issue by danielFesenmeyer** [keycloak/keycloak#17679]
### Describe the bug

This ticket is for aligning the Admin UI with the implementation done in this Keycloak issue: https://github.com/keycloak/keycloak/issues/15845

**IMPORTANT**: The keycloak issue should be merged first, otherwise the linked PR will NOT completely work.

It would be nice to have the exactly same fallback implementation as for the other themes, but unluckily the i18next library (https://github.com/i18next/i18next) seems NOT to support variants (like "de-CH-*1996*"), just regions (like "de-CH") and languages without region (like "de"), of course.
If variants should be supported, that feature could be implemented as a PR for i18next - or maybe i18next's language selection hook could be implemented accordingly.

The fallback priorities implemented in the linked PR are as follows (note that variant is NOT supported yet):

RL \<language-region> **>** T \<language-region> **>** RL \<language> **>** T \<language> **>** RL en **>** T en

(Abbreviations: RL = realm localization, T = Theme i18n files)

### Version

20.0.3

### Environment

all environments

### Expected behavior

GIVEN the supported locales for the realm are "de-CH", "de" and "en"
AND realm localization texts with key "common:realmSettings" are defined for the locales "de" and "en"
AND the user has selected the language "de-CH"
WHEN the user selects any page in the UI
THEN the sidebar shows the "realm settings" text for locale "de" 

### Actual behavior

GIVEN the supported locales for the realm are "de-CH", "de" and "en"
AND realm localization texts with key "realmSettings" are defined for the locales "de" and "en"
AND the user has selected the language "de-CH"
WHEN the user selects any page in the UI
THEN the sidebar shows the "realm settings" text for locale "en" 

It's incorrect that the "en" translation is shown, because it's less specific than the "de" translation.

Also note that the localization text keydoes not contain the namespace prefix "common:". Currently whether the prefix has to be part of the localization text key, depends on how it is called from the react code (with or without the default "common:" prefix). I think it's obvious that the namespace must always be part of the localization text key - or not at all.

### How to Reproduce?

see behavior description above.

### Anything else?

Precondition: https://github.com/keycloak/keycloak/issues/15845 has to be merged

**Comment by danielFesenmeyer** [comment]
Implementation has been done in context of https://github.com/keycloak/keycloak/issues/20182 and https://github.com/keycloak/keycloak/issues/15845
